src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,27 @@
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const { query, status } = useAppSelector(state => state.filter);
+  const dispatch = useAppDispatch();
+
+  const setStatus = (newStatus: Status) => {
+    dispatch(filterActions.setStatus(newStatus));
+  };
+
+  const setQuery = (text: string) => dispatch(filterActions.setQuery(text));
+
   return (
-    <form
-      className="field has-addons"
-      onSubmit={event => event.preventDefault()}
-    >
+    <form className="field has-addons">
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select
+            data-cy="statusSelect"
+            value={status}
+            onChange={(event) => setStatus(event.target.value as Status)}
```

Comment:
It's better to move handlers to separate functions than to create them right inside attributes.